src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ExtractToCodeBehindCodeActionResolver.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using CSharpSyntaxFactory = Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal class ExtractToCodeBehindCodeActionResolver : RazorCodeActionResolver
+    {
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly DocumentResolver _documentResolver;
+        private readonly FilePathNormalizer _filePathNormalizer;
+        private readonly ILogger _logger;
+
+        private static readonly Range StartOfDocumentRange = new Range(new Position(0, 0), new Position(0, 0));
+
+        public ExtractToCodeBehindCodeActionResolver(
+            ForegroundDispatcher foregroundDispatcher,
+            DocumentResolver documentResolver,
+            FilePathNormalizer filePathNormalizer,
+            ILoggerFactory loggerFactory)
+        {
+            if (foregroundDispatcher is null)
+            {
+                throw new ArgumentNullException(nameof(foregroundDispatcher));
+            }
+
+            if (documentResolver is null)
+            {
+                throw new ArgumentNullException(nameof(documentResolver));
+            }
+
+            if (filePathNormalizer is null)
+            {
+                throw new ArgumentNullException(nameof(filePathNormalizer));
+            }
+
+            if (loggerFactory is null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _foregroundDispatcher = foregroundDispatcher;
+            _documentResolver = documentResolver;
+            _filePathNormalizer = filePathNormalizer;
+            _logger = loggerFactory.CreateLogger<ExtractToCodeBehindCodeActionProvider>();
+        }
+
+        public override string Action => LanguageServerConstants.CodeActions.ExtractToCodeBehindAction;
+
+        public override async Task<WorkspaceEdit> ResolveAsync(JObject data, CancellationToken cancellationToken)
```

Comment:
the namign of async methods seems inconsistent.  Some end with `Async`, some do not (like `Handle`).

Reply:
Ya sadly that's a convention enforced upon us by the underlying OmniSharp framework that we depend on for our OOP langauge server